scripts/cleanOutdatedFrameworkProductions.py

```diff
@@ -0,0 +1,202 @@
+CMSSW_BASE = os.environ['CMSSW_BASE']
+SCRAM_ARCH = os.environ['SCRAM_ARCH']
+sys.path.append(os.path.join(CMSSW_BASE,'bin', SCRAM_ARCH))
+
+# Add default ingrid storm package
+sys.path.append('/nfs/soft/python/python-2.7.5-sl6_amd64_gcc44/lib/python2.7/site-packages/storm-0.20-py2.7-linux-x86_64.egg')
+sys.path.append('/nfs/soft/python/python-2.7.5-sl6_amd64_gcc44/lib/python2.7/site-packages/MySQL_python-1.2.3-py2.7-linux-x86_64.egg')
+
+from SAMADhi import Dataset, Sample, File, DbStore
+
+def get_options():
+    """
+    Parse and return the arguments provided by the user.
+    """
+    username = getpwuid(os.stat(os.getcwd()).st_uid).pw_name
+    parser = argparse.ArgumentParser(description='Provide a list of things to be deleted in /storage/data/cms/store/user/')
+    parser.add_argument('--crabUsername', action='store', dest='crabUsername', default=username, type=str,
+        help='crab / storage username')
+    parser.add_argument('--ingridUsername', action='store', dest='ingridUsername', default=username, type=str,
+        help='ingrid username')
+    parser.add_argument('--debug', action='store_true', help='debug mode: running on only one production and do not delete anything', dest='debug')
+    options = parser.parse_args()
+    return options
+
+def main(crabUsername, ingridUsername, DEBUG = False):
+    if DEBUG:
+        print "RUNNING IN DEBUG MODE"
+        print "Nothing will be deleted\n"
+
+    dbstore = DbStore()
+
+    print "##### Get the list of potential DB samples of interest"
+    list_allDBsamples = []
+    results = dbstore.find(Sample)
+    for r in results:
+        if r.author is None:
+            continue
+        if crabUsername in r.path or ingridUsername in r.author:
+            if r.path == '':
+                continue
+            list_allDBsamples.append([r.name, r.source_dataset_id])
+    print ""
+
+    print "##### Get the list of existing productions"
+    # before anything else: get the list of tags to not touch
+    whitelist = requests.get('https://raw.githubusercontent.com/cp3-llbb/GridIn/master/data/SAMADhi_doNOTdelete_whitelist.json').json()
+    print whitelist
+    list_all_productions = []
+    for i, s in enumerate(list_allDBsamples):
+        s_name, s_id = s
+        isProdAlreadyListed = False
+        isSampleProtected = False
+        for FWtag, Anatag in list_all_productions:
+            if FWtag in str(s_name) and Anatag in str(s_name):
+#                print "This prod is already in the list, FWtag= ", FWtag, "Anatag= ", Anatag
+                isProdAlreadyListed = True
+                break
+        if isProdAlreadyListed:
+            continue
+        result = dbstore.find(Dataset, Dataset.dataset_id == s_id)
+        p_name = list(result.values(Dataset.process))[0]
+        tags = str(s_name)
+        # Get the tags: 
+        # First of all: check if the sample is protected or not
+        for part in ['HHAnalysis', 'TTAnalysis', 'ZAAnalysis', 'TTWAnalysis']:
```

Comment:
But if you iterate over the dict, you'd still have the key and the value, so you can still remove the analyzer name from the tag